recipe/emailverification/emailverificationClaim.go

```diff
@@ -0,0 +1,63 @@
+package emailverification
+
+import (
+	"errors"
+	"time"
+
+	evclaims "github.com/supertokens/supertokens-golang/recipe/emailverification/claims"
+	"github.com/supertokens/supertokens-golang/recipe/session/claims"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+// key string, fetchValue claims.FetchValueFunc
+func NewEmailVerificationClaim() *evclaims.TypeEmailVerificationClaim {
+	fetchValue := func(userId string, userContext supertokens.UserContext) (interface{}, error) {
+		instance, err := getRecipeInstanceOrThrowError()
+		if err != nil {
+			return nil, err
+		}
+		emailInfo, err := instance.GetEmailForUserID(userId, userContext)
+		if err != nil {
+			return false, err
+		}
+		if emailInfo.OK != nil {
+			verified, err := (*instance.RecipeImpl.IsEmailVerified)(userId, emailInfo.OK.Email, userContext)
+			if err != nil {
+				return false, nil
+			}
+			return verified, nil
+		} else if emailInfo.EmailDoesNotExistError != nil {
+			// We consider people without email addresses as validated
+			return true, nil
+		} else {
+			return false, errors.New("should never come here: UnknownUserIdError or invalid result from getEmailForUserId")
+		}
+	}
+
+	booleanClaim := claims.BooleanClaim("st-ev", fetchValue, nil)
+
+	emailVerificationClaim := &evclaims.TypeEmailVerificationClaim{
+		TypeBooleanClaim: booleanClaim,
+	}
+	emailVerificationClaim.Validators = &evclaims.EmailVerificationClaimValidators{
+		BooleanClaimValidators: booleanClaim.Validators,
+		IsVerified: func(refetchTimeOnFalseInSeconds *int64) *claims.SessionClaimValidator {
+			if refetchTimeOnFalseInSeconds == nil {
+				var defaultTimeout int64 = 10
+				refetchTimeOnFalseInSeconds = &defaultTimeout
+			}
+
+			claimValidator := booleanClaim.Validators.HasValue(true, nil, nil)
+			claimValidator.ShouldRefetch = func(payload map[string]interface{}, userContext supertokens.UserContext) bool {
+				value := emailVerificationClaim.GetValueFromPayload(payload, userContext)
+				return value == nil || (value == false && *emailVerificationClaim.GetLastRefetchTime(payload, userContext) < time.Now().UnixMilli()-*refetchTimeOnFalseInSeconds*1000)
+			}
```

Comment:
Technically it is not same because, only false case checks on the last fetch time and ignored on true. That way, true and false gets different lifetime. But in this context I don't know why we need such a behaviour.

Reply:
The reason is that if email verification is false, we want to query the isEmailVerified API each time until it has been verified so that when the user verifies their email, the session knows about it immediately.